storage/sql/coresql/testonly/checks.go

```diff
@@ -17,34 +17,35 @@ package testonly
 import (
 	"bytes"
 	"fmt"
-	"testing"
 
 	"github.com/google/trillian"
 	"github.com/google/trillian/storage"
 )
 
-// CheckLeafContents compares two log leaves to see if they match. It fails a test with
-// a descriptive message if not.
-func CheckLeafContents(leaf *trillian.LogLeaf, seq int64, rawHash, hash, data, extraData []byte, t *testing.T) {
+// CheckLeafContents compares two log leaves to see if they match. Returns nil if they do
+// otherwise a descriptive error.
+func CheckLeafContents(leaf *trillian.LogLeaf, seq int64, rawHash, hash, data, extraData []byte) error {
 	if got, want := leaf.MerkleLeafHash, hash; !bytes.Equal(got, want) {
-		t.Fatalf("Wrong leaf hash in returned leaf got\n%v\nwant:\n%v", got, want)
+		return fmt.Errorf("wrong leaf hash in returned leaf got\n%v\nwant:\n%v", got, want)
 	}
 
 	if got, want := leaf.LeafIdentityHash, rawHash; !bytes.Equal(got, want) {
-		t.Fatalf("Wrong raw leaf hash in returned leaf got\n%v\nwant:\n%v", got, want)
+		return fmt.Errorf("Wrong raw leaf hash in returned leaf got\n%v\nwant:\n%v", got, want)
```

Comment:
Go frees you from tedious debates about how code should be formatted, so you can spend the time on tedious debates about how error messages should be formatted. I'll have another look at them.